src/nccl_ofi_rdma.c

```diff
@@ -5762,19 +5810,29 @@ static nccl_net_ofi_rdma_device_rail_t *create_device_rail_array(struct fi_info
 	}
 
 	for (int i = 0 ; i < num_infos ; i++) {
+		int vf_idx, rail_idx;
+
 		if (info_list == NULL) {
 			goto error;
 		}
 
+		vf_idx = get_rail_vf_idx(info_list);
+		if (vf_idx == -1) {
+			rail_idx = i;
+		} else {
+			rail_idx = rail_map[vf_idx];
+			rail_map[vf_idx]++;
+		}
```

Comment:
I think we should restrict this change to only the EFA provider. (Or, just allow it to fail on other providers, but I'm not ecstatic about that option.)

Reply:
Why?

Reply:
It relies on a GUID available in `/sys/class/infiniband/%s/node_guid` as well as assumes a particular GUID format to locate vf_idx (https://github.com/aws/aws-ofi-nccl/pull/438/commits/3d1d05f25273fb04648ce71e0a6791d5351a83a7#diff-70b13370752f15bd8052cf01a6f9748fdd022b830a4f0aeacc10add288008540R5785-R5789), both of which I thought are EFA-specific.

Reply:
and that's what I get for trying to look on my phone.  You're right, this needs to be in platform-aws.

Reply:
Done